index.ts

```diff
@@ -53,8 +53,14 @@ type Method = (this: MessengerMeta, ...args: Arguments) => Promise<unknown>;
 
 // TODO: It may include additional meta, like information about the original sender
 type Message<TArguments extends Arguments = Arguments> = {
-  type: string;
+  type: keyof MessengerMethods;
   args: TArguments;
+
+  /** If the message is being sent to an intermediary receiver, also set the target */
+  target?: Target;
+
+  /** If the message is being sent to an intermediary receiver, also set the options */
+  options?: Target;
```

Comment:
Notification forwarding hasn't been implemented yet, so this is currently unused. I'll implement them separately

https://github.com/pixiebrix/webext-messenger/issues/31